MovingBlocks/Terasology
# feat(subsystems): Extract GsonSerializer

### Contains

GsonSerializer as Subsystem.

Require https://github.com/MovingBlocks/Terasology/pull/4255 first.

### How to test
1. engine-tests works.

2. Open Terasology.
3. Check Nui elements (part of GsonSerializer used by skins  and ui assets)
4. Check Audio config (part of GsonSerializer used by AutoConfigs)
5. Check Record & Replay (GsonSerializer used by this fully)

### Follow ups

[ ] - migrate engine's classes from usage `Gson` to usage `GsonSerializer`
